crates/storage/src/collections/hashmap/impls.rs

```diff
@@ -45,7 +45,15 @@ where
     Key: From<<H as HashOutput>::Type>,
 {
     fn drop(&mut self) {
-        self.clear_cells();
+        if self.values.key().is_none() {
+            // We won't clear any storage if we are in lazy state since there
+            // probably has not been any state written to storage, yet.
+            return
+        }
```

Comment:
Not connected to this PR really but still a very needed fix. How was this not discovered until now?